Gracefully handle initialization errors in oracle and monitor

da-kami opened the pull request:
fixes #1848 

If we fail to process one CFD we should not break the loop because we might not process the other CFDs leading to problems.
We don't fail the application if we run into an error with initializing the monitor or the oracle, so we should handle the initialization errors inside the actor.
I think fallible tasks when starting an actor that is not supervised should be avoided because we don't handle the errors and it is unclear what the behavior of the application would be.

luckysori commented on daemon/src/monitor.rs:
```diff
@@ -554,77 +554,77 @@ impl xtra::Actor for Actor {
+                                kind: TransactionKind::Commit,
+                            })
+                            .await
+                            .expect("receiver to outlive the actor")
```
❓ I didn't expect to see this change in this commit. Can we undo it or at least move it to a separate commit?

da-kami replied:
After having another look I am actually not sure we *should* expect in these cases. 

These are cases where we send to ourselves from a task that we spawn (and for some cases actually react to the result that the handler returns). I think we can assume that the actor should always be alive in such a case (because the task is cleaned up with the actor).
But probably it's still better to treat this as it was. I removed the expects - which means that in `monitor` the tasks added remain fallible.

luckysori reviewed (approved):
Good job using `next` instead of `try_next`!

I would like to see the changes from `?` to `expect` in a separate commit.

da-kami committed:
Gracefully handle initialization errors in oracle and monitor

If we fail to process one CFD we should not break the loop because we might not process the other CFDs leading to problems.
We don't fail the application if we run into an error with initializing the monitor or the oracle, so we should handle the initialization errors inside the actor.
Fallible tasks when starting an actor that is not supervised should be avoided.

klochowicz reviewed (approved):
well spotted! there is no reason to stop loading other CFDs if one fails.

bonomat commented on daemon/src/oracle.rs:
```diff
@@ -312,37 +312,33 @@ impl xtra::Actor for Actor {
                         }
+                    };
+                    if let Some(pending_attestation) = pending_attestation {
+                        let _: Result<(), xtra::Disconnected> = this
```
Do you need the type hint if you ignore it? I see that it has been there before, but maybe it is unnecessary.

bonomat reviewed (commented):
LGTM, thanks for fixing this.
